libs/community/langchain_community/callbacks/uptrain_callback.py

```diff
@@ -0,0 +1,390 @@
+        # If both are entered, the UpTrain API key is used
+        if key_type == "uptrain":
+            settings = Settings(uptrain_access_token=api_key)
+            self.uptrain_client = APIClient(settings=settings)
+        elif key_type == "openai":
+            settings = Settings(openai_api_key=api_key, evaluate_locally=False)
+            self.uptrain_client = EvalLLM(settings=settings)
+        else:
+            raise ValueError("Invalid key type: Must be 'uptrain' or 'openai'")
+
+            
+        
+    def uptrain_evaluate(
+        self,
+        project_name: str,
+        data: List[Dict[str, str]],
+        checks: List[str],
+    ) -> None:
+        """Run an evaluation on the UpTrain server using UpTrain client."""
+        if self.uptrain_client.__class__.__name__ == "APIClient":
+            uptrain_result = self.uptrain_client.log_and_evaluate(
+                project_name=project_name,
+                data=data,
+                checks=checks,
+            )
+        else:
+            uptrain_result = self.uptrain_client.evaluate(
+                data=data,
+                checks=checks,
+            )
+        self.schema.uptrain_results[project_name].append(uptrain_result)
+
+        score_name_map = {
+            "score_context_relevance": "Context Relevance Score",
+            "score_factual_accuracy": "Factual Accuracy Score",
+            "score_response_completeness": "Response Completeness Score",
+            "score_sub_query_completeness": "Sub Query Completeness Score",
+            "score_context_reranking": "Context Reranking Score",
+            "score_context_conciseness": "Context Conciseness Score",
+            "score_multi_query_accuracy": "Multi Query Accuracy Score",
+        }
+
+        for row in uptrain_result:
+            columns = list(row.keys())
+            for column in columns:
+                if column == "question":
+                    print(f"\nQuestion: {row[column]}")
+                    self.first_score_printed_flag = False
+                elif column == "response":
+                    print(f"Response: {row[column]}")
+                    self.first_score_printed_flag = False
+                elif column == "variants":
+                    print(f"Multi Queries:")
+                    for variant in row[column]:
+                        print(f"  {variant}")
+                    self.first_score_printed_flag = False
+                elif column.startswith("score"):
+                    if not self.first_score_printed_flag:
+                        print()
+                        self.first_score_printed_flag = True
+                    if column in score_name_map:
+                        print(f"{score_name_map[column]}: {row[column]}")
+                    else:
+                        print(f"{column}: {row[column]}")
+            print()
+
+    def on_llm_start(
```

Comment:
The sync methods should not exist on this callback handler. The callback handler inherits from an async callback handler and should only implement the async methods that are needed

Reply:
Changed it to BaseCallbackHandler